pkg/planner/cascades/memo/group.go

```diff
@@ -45,7 +45,7 @@ type Group struct {
 
 	// hash2ParentGroupExpr is reverted pointer back from current Group to parent referred GEs.
 	// uint64 means *list.Element's addr, list.Element means the pos in global memo group expression's list.
-	hash2ParentGroupExpr *hashmap.Map[*GroupExpression, struct{}]
+	hash2ParentGroupExpr *hashmap.Map[*GroupExpression, bool]
```

Comment:
the generic hashmap will error as debug log when val type are struct{} which is interface type. so i change it to bool